overlays/haskell.nix

```diff
@@ -551,10 +552,50 @@ final: prev: {
                 ) rawProject.hsPkgs
                 // {
                   # These are functions not packages
-                  inherit (rawProject.hsPkgs) shellFor makeConfigFiles ghcWithHoogle ghcWithPackages;
+                  inherit (rawProject.hsPkgs) makeConfigFiles ghcWithHoogle ghcWithPackages;
+                  shellFor = a: __trace "project.hsPkgs.shellFor has been deprecated in favour of project.shellFor"
+                    (rawProject.hsPkgs.shellFor a);
               });
 
             projectCoverageReport = haskellLib.projectCoverageReport (map (pkg: pkg.coverageReport) (final.lib.attrValues (haskellLib.selectProjectPackages hsPkgs)));
+
+            cross = (final.lib.mapAttrs (_: pkgs:
+                rawProject.projectFunction pkgs.haskell-nix
```

Comment:
This is going to be weird, right? We'll end up with `project.cross.x.cross.x.cross.x....` or something?

Reply:
It will behave like `pkgs.pkgsCross.x.pkgsCross.x.pkgsCross.x....`.